rcsbapi/data/data_query.py

```diff
@@ -97,6 +97,9 @@ def _process_input_ids(self, input_type: str, input_ids: Union[List[str], Dict[s
                 # If not converted, retrieve id list from dictionary
                 input_ids = list(input_ids[DATA_SCHEMA._root_dict[input_type][0]["name"]])
 
+        # Make all input_ids uppercase
+        input_ids = [id.upper() for id in input_ids]
```

Comment:
This is a good failsafe to include for handling the case-sensitivity issue, but it's more of a patch than a fix. In addition to this, I would also adjust the regex patterns defined in `data_schema.py` to also accept lowercase in addition to uppercase characters (e.g., using `[A-Za-z0-9]` rather than just `[A-Z0-9]`)). 

Better yet, it would be ideal if these regex patterns could be defined in the `const.py` file itself, and then used by the `data_schema.py`...

Reply:
I ended up editing the regex strings so they reflect what's allowed in the API and then moved them to the const file. Just so if we move the regex checks to occur before ids are capitalized, it will raise an ValueError.